modules/core/src/smithy4s/http/HttpRequest.scala

```diff
@@ -96,6 +96,28 @@ object HttpRequest {
         req.addHeaders(meta.headers).copy(uri = newUri)
     }
 
+    private[http] def hostPrefix[Body, I](
+        httpEndpoint: OperationSchema[I, _, _, _, _]
+    ): Writer[Body, I] = {
+      HostPrefix(httpEndpoint) match {
+        case Some(prefixEncoder) =>
+          new Writer[Body, I] {
+            def write(
+                request: HttpRequest[Body],
+                input: I
+            ): HttpRequest[Body] = {
+              val hostPrefix = prefixEncoder.write(List.empty, input)
+              val oldUri = request.uri
+              val newUri =
+                oldUri.copy(host = s"${hostPrefix.mkString("")}.${oldUri.host}")
```

Comment:
there is no separator between the prefix parts? no `.`? anyhow, I think you could discard the empty string and just do `.mkString` if it is no separator

Reply:
a test just to see the effect on the uri would be valuable I think

Reply:
Separators are already put into those segments (as they can be anything as defined in the endpoint trait), but yeah I will definitely look at adding a test there (currently is covered by compliance tests also)